packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -633,7 +633,7 @@ abstract class ResidentHandlers {
   /// the value of [fullRestart].
   Future<OperationResult> restart({ bool fullRestart = false, bool pause = false, String reason }) {
     final String mode = isRunningProfile ? 'profile' :isRunningRelease ? 'release' : 'this';
-    throw '${fullRestart ? 'Restart' : 'Reload'} is not supported in $mode mode';
+    throw UnsupportedError('${fullRestart ? 'Restart' : 'Reload'} is not supported in $mode mode');
```

Comment:
This is another case where checking what is happening at the call site is a good idea. `UnsupportedError` will cause the tool to crash, whereas the call site might be trying to emit the error string and continuing.

Reply:
I didn't realize the distinction between Error and Exception. I've changed this to `Exception`.

Comment about checking the callsite in another comment below.